src/openforms/tests/search_strategies.py

```diff
@@ -0,0 +1,30 @@
+from string import ascii_letters
+
+from django.core.exceptions import ValidationError
+
+from hypothesis import strategies as st
+
+from openforms.forms.models.form_variable import variable_key_validator
+from openforms.typing import JSONPrimitive
+
+json_primitives: st.SearchStrategy[JSONPrimitive]
+json_primitives = st.one_of(
+    st.none(),
+    st.booleans(),
+    st.integers(),
+    st.floats(allow_infinity=False, allow_nan=False),
+    st.text(),
+)
```

Comment:
Maybe move the `json_collections` and `json_values` here too.

`field_names` is pretty tied to the `HeaderValidator` with 0 reuse value.